pkg/clustermesh/remote_cluster.go

```diff
@@ -175,7 +175,7 @@ func (rc *remoteCluster) Status() *models.RemoteCluster {
 }
 
 func (rc *remoteCluster) ClusterConfigRequired() bool {
-	return rc.mesh.conf.ConfigValidationMode == types.Strict
+	return (rc.mesh.conf.ConfigValidationMode == types.Strict) || rc.mesh.conf.ClusterInfo.ExtendedClusterMeshEnabled()
```

Comment:
The `ClusterInfo.ExtendedClusterMeshEnabled` part could be directly moved into `ValidateRemoteConfig`, as it only depends on `ClusterInfo`.

Reply:
This is true, but I think it might make more sense here if we expect someone may reuse ClusterConfigRequired again in the future. If extended clustermesh is enabled, the config will always be required

Reply:
ending up going with your suggestion, as after moving the test it introduced a cyclical dependency :grin: